syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.CSTB.message";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package message;

service Consumer {
  rpc Subscribe (AskForSubscribe) returns (AskForSubscribeReplay) {}
  rpc ListQuees (AskForQuees) returns (AskForQueesReplay) {}
  rpc ConsumeMessage (AskForMessage) returns (AskForMessageReplay) {}
}

message AskForSubscribe {
  string channel = 1;
  string consumerName = 2;
}

message AskForSubscribeReplay {
  string ack = 1;
}

message AskForQuees {
  string ask = 1;
}

message AskForQueesReplay {
  repeated string listQuees = 1;
}

message AskForMessage {
  string channel = 1;
  string consumerName = 2;
}

message AskForMessageReplay {
  string ack = 1;
}